internal/cmd/todos/app.go

```diff
@@ -200,34 +177,27 @@ func newTODOsApp() *cli.App {
 			if err != nil {
 				return err
 			}
-			w := walker.New(opts)
-			if w.Walk() {
-				return ErrWalk
-			}
 
-			return nil
-		},
-		ExitErrHandler: func(c *cli.Context, err error) {
-			if err == nil {
-				return
+			todosFound := false
+			todoFunc := opts.TODOFunc
+			opts.TODOFunc = func(o *walker.TODORef) error {
+				todosFound = true
+				return todoFunc(o)
 			}
 
-			// NOTE: Walk errors return an exit code but do not print the error as it
-			// has presumably already been handled.
-			if errors.Is(err, ErrWalk) {
-				cli.OsExiter(ExitCodeWalkError)
-				return
+			w := walker.New(opts)
+			if w.Walk() {
+				return ErrWalk
 			}
 
-			// ExitCode return an exit code for the given error.
-			_ = utils.Must(fmt.Fprintf(c.App.ErrWriter, "%s: %v\n", c.App.Name, err))
-			if errors.Is(err, ErrFlagParse) {
-				cli.OsExiter(ExitCodeFlagParseError)
-				return
+			if todosFound {
+				return ErrTODOsFound
 			}
 
-			cli.OsExiter(ExitCodeUnknownError)
+			return nil
```

Comment:
Need a test of `app.Action` for when todos are found and when todos are not found.